app/policies/default_agent_policy.rb

```diff
@@ -55,7 +55,7 @@ def same_service?
     elsif @record.respond_to?(:agent_id)
       @record.agent.service_id == @context.agent.service_id
     elsif @record.respond_to?(:agent_ids)
-      @record.agents.where(agents: { service_id: @context.agent.service_id }).exists?
+      Agent.where(id: @record.agent_ids).pluck(:service_id).include?(@context.agent.service_id)
```

Comment:
I don't think I see the catch here, what didn't work in the old version?

Reply:
```
@record.persisted?
=> false

>> Rdv.last.agents
=> #<ActiveRecord::Associations::CollectionProxy [#<Agent id: 1, email: "agent@gmail.com", created_at: "2020-03-31 09:15:56", updated_at: "2020-04-07 07:56:53", role: "admin", first_name: "Ethan", last_name: "MÉNARD", deleted_at: nil, service_id: 1>]>

>> Rdv.last.agents.where(service_id: 1)
=> #<ActiveRecord::AssociationRelation [#<Agent id: 1, email: "agent@gmail.com", created_at: "2020-03-31 09:15:56", updated_at: "2020-04-07 07:56:53", role: "admin", first_name: "Ethan", last_name: "MÉNARD", deleted_at: nil, service_id: 1>]>

>> @record.agents
=> #<ActiveRecord::Associations::CollectionProxy [#<Agent id: 1, email: "agent@gmail.com", created_at: "2020-03-31 09:15:56", updated_at: "2020-04-07 07:56:53", role: "admin", first_name: "Ethan", last_name: "MÉNARD", deleted_at: nil, service_id: 1>]>

>> @record.agents.where(service_id: 1)
=> #<ActiveRecord::AssociationRelation []>
```

Le @record n'étant pas persisté le .where ne fonctionnait pas.

Reply:
👌 got it bien vu merci !

Reply:
je pinaille, mais ne serait-ce pas plus safe de checker que tous les agents sont du même service ? sinon tu peux (en théorie) créer un RDV associé à  un agent de ton service et un autre agent d'un autre service

Reply:
`Agent.where(id: @record.agent_ids).pluck(:service_id).uniq == [@context.agent.service_id]`